src/controllers/middleware.js

```diff
@@ -1,36 +1,25 @@
 // MIDDLEWARE to ensure user is authenticated
-exports.updateUserSession = (req, res, next) => {
-  let userInfo;
+exports.ensureAuthenticated = (req, res, next) => {
   if (req.isAuthenticated()) {
-    userInfo = req.user;
-    if (req.session.registeredProfile) {
-      userInfo.session = {
-        login: true,
-        signup: false,
-      };
-      return next(null, userInfo);
-    }
+    return next();
+  }
+  res.redirect('/');
+};
+
+// Updates user object to include user session
+exports.updateUserSession = (req) => {
```

Comment:
It is a bit confusing that we have `req.session` and userInfo.session`

The name of this function, at first, seemed to suggest to me that we want to the change session object that belongs to the argument provided.

Reply:
Good point, the name of the function is confusing